### ISSUE TITLE
ERROR:localstack.services.generic_proxy: Error forwarding request:

### ISSUE THREAD

**Issue by mmraucorp** [localstack/localstack#894]
Starting local dev environment. CTRL-C to quit.
Starting mock API Gateway (http port 4567)...
Starting mock DynamoDB (http port 4569)...
Starting mock SES (http port 4579)...
Starting mock Kinesis (http port 4568)...
Starting mock Redshift (http port 4577)...
Starting mock S3 (http port 4572)...
Starting mock CloudWatch (http port 4582)...
Starting mock CloudFormation (http port 4581)...
Starting mock SSM (http port 4583)...
Starting mock SQS (http port 4576)...
Starting local Elasticsearch (http port 4571)...
Starting mock SNS (http port 4575)...
Starting mock DynamoDB Streams service (http port 4570)...
Starting mock Firehose service (http port 4573)...
Starting mock Route53 (http port 4580)...
Starting mock ES service (http port 4578)...
Starting mock Lambda service (http port 4574)...
2018-08-11T13:33:08:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d442415d0>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
  File "/home/maruf/.local/lib/python2.7/site-packages/localstack/services/generic_proxy.py", line 201, in forward
    headers=forward_headers)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d442415d0>: Failed to establish a new connection: [Errno 111] Connection refused',))

2018-08-11T13:34:08:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4425fa10>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
  File "/home/maruf/.local/lib/python2.7/site-packages/localstack/services/generic_proxy.py", line 201, in forward
    headers=forward_headers)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4425fa10>: Failed to establish a new connection: [Errno 111] Connection refused',))

2018-08-11T13:35:09:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4429c3d0>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
  File "/home/maruf/.local/lib/python2.7/site-packages/localstack/services/generic_proxy.py", line 201, in forward
    headers=forward_headers)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4429c3d0>: Failed to establish a new connection: [Errno 111] Connection refused',))

2018-08-11T13:36:09:ERROR:localstack.services.generic_proxy: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4425f910>: Failed to establish a new connection: [Errno 111] Connection refused',)) Traceback (most recent call last):
  File "/home/maruf/.local/lib/python2.7/site-packages/localstack/services/generic_proxy.py", line 201, in forward
    headers=forward_headers)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/maruf/.local/lib/python2.7/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=4564): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4d4425f910>: Failed to establish a new connection: [Errno 111] Connection refused',))

**Comment by aw230012** [comment]
Also receiving this error.

**Comment by codeinaire** [comment]
I'm receiving this error as well. I'm using a MacBook Pro with High Sierra v10.13.6. I'm using the localstack docker image in a docker compose file like this:

```
version: '3.4'
services:
  localstack:
    image: localstack/localstack
    environment:
      SERVICES: "lambda,sqs,sns,dynamodb,s3"
      LAMBDA_EXECUTOR: docker-reuse
      DEFAULT_REGION: ap-southeast-2
    ports:
      - 4572:4572
      - 4569:4569
      - 4575:4575
      - 4576:4576
      - 4574:4574
      - 4564:4564
      - 8080:8080
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
  db:
    image: postgres
  web:
    build: .
    command: bundle exec rerun --background "rackup -p 5678 -o '0.0.0.0'"
    depends_on:
      - db
      - localstack
    entrypoint: ''  # to prevent chamber running in CI.
    environment:
      RAILS_ENV: development
      RACK_ENV: development
      PGHOST: db
      PGUSER: postgres
      NEW_RELIC_AGENT_ENABLED: 'false'
      LAMBDA_ARCHIVE_CREATE: CreateArchive
      LAMBDA_ARCHIVE_INDEX: ListArchive
      AWS_ACCESS_KEY_ID: whatever
      AWS_SECRET_ACCESS_KEY: test
      TOPIC_ARN: arntest
      AWS_REGION: ap-southeast-2
    ports:
      - 5678:5678
    volumes:
      - .:/app
      - ${PWD}/config/database-prod.yml:/app/config/database.yml
```

If I take out using port 4564, I still get the error.

**Comment by manypozo** [comment]
Is this a duplicated issue? [Issue 241](https://github.com/localstack/localstack/issues/241)
As stated, it seems that wiping the infra folder would allow to nicely start docker.
In addition, it is recommended to start the docker container from `localstack start --docker` instead of directly using the docker compose. 
The wiping + localstack start --docker work fine for me.
Do not forget to update your docker image though.

**Comment by whummer** [comment]
Should be fixed in the meantime. Please re-open if the problem persists. Thanks

**Comment by martinbliss** [comment]
@whummer Im able to replicate this issue using two recent docker images available on Docker Hub: `6e1c4f42467d` (latest as of now) and `0.10.6`

Full Disclosure: I haven't tried jumping into the container to try and wipe the infra folder (as suggested earlier in this thread). I'm simply looking to see if this fix has made its way into the publicly available Docker images. 

Steps (using `awslocal` CLI):
1. Create an S3 bucket
2. Copy CFN template to bucket
3. Issue `awslocal cloudformation create-stack --stack-name (abc) --template-url s3://(bucket)/(cfn file)`

Dump:

```
2020-02-16T18:14:56:ERROR:localstack.services.generic_proxy: Error forwarding request: 'NoneType' object has no attribute 'status_code' Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 242, in forward
    path=path, data=data, headers=forward_headers)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_listener.py", line 159, in forward_request
    modified_request = transform_template(req_data)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_listener.py", line 70, in transform_template
    template_body = get_template_body(req_data)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_listener.py", line 113, in get_template_body
    raise Exception('Unable to fetch template body (code %s) from URL %s' % (response.status_code, url))
AttributeError: 'NoneType' object has no attribute 'status_code'
```
